src/component/DefaultLegendContent.tsx

```diff
@@ -104,7 +106,8 @@ class DefaultLegendContent<TValue, TID> extends PureComponent<Props<TValue, TID>
       );
     }
     if (React.isValidElement(data.legendIcon)) {
-      return React.cloneElement(data.legendIcon);
+      const iconProps: any = _.omit(data, 'legendIcon');
+      return React.cloneElement(data.legendIcon, iconProps);
```

Comment:
please do not use lodash omit for that.
* it is slow
* it has been removed from lodash 5.x

consider this approach

```
      const iconProps: any = { ...data };
      delete iconProps.legendIcon;
```

Reply:
Thank you for your review! I modified that.